langlib/lang.array/src/main/java/org/ballerinalang/langlib/array/utils/ArrayUtils.java

```diff
@@ -101,26 +102,31 @@ public static BError createOpNotSupportedError(Type type, String op) {
     public static BArray createEmptyArrayFromTuple(BArray arr) {
         Type arrType = TypeUtils.getImpliedType(arr.getType());
         TupleType tupleType = (TupleType) arrType;
-        List<Type> memTypes = new ArrayList<>();
         List<Type> tupleTypes = tupleType.getTupleTypes();
+        Type restType = tupleType.getRestType();
         boolean isSameType = true;
         Type sameType = null;
```

Comment:
```suggestion
        Type firstType = null;
```
Having the name `sameType` is not good. What you assign here is the first type.

Reply:
Removed these logic.